tests/foreman/api/test_docker.py

```diff
@@ -856,6 +856,229 @@ def test_positive_promote_multiple_with_docker_repo_composite(self):
             promote(comp_cvv, lce.id)
             self.assertEqual(len(comp_cvv.read().environment), i + 1)
 
+    @tier2
+    @upgrade
+    @run_only_on('sat')
+    def test_positive_name_pattern_change(self):
+        """Promote content view with Docker repository to lifecycle environment.
+        Change registry name pattern for that environment. Verify that repository
+        name on product changed according to new pattern.
+
+        :id: cc78d82d-027b-4cb7-92c5-dcccf9b592ea
+
+        :expectedresults: Container repository name is changed
+            according to new pattern.
+
+        :CaseLevel: Integration
+        """
+        pattern_prefix = gen_string('alpha', 5)
+        docker_upstream_name = 'hello-world'
+        new_pattern = ("{}-<%= organization.label %>"
+                       "/<%= repository.docker_upstream_name %>").format(
+                pattern_prefix)
+
+        repo = _create_repository(
+                entities.Product(organization=self.org).create(),
+                upstream_name=docker_upstream_name)
+        repo.sync()
+        content_view = entities.ContentView(
+                composite=False,
+                organization=self.org,
+        ).create()
+        content_view.repository = [repo]
+        content_view = content_view.update(['repository'])
+        content_view.publish()
+        cvv = content_view.read().version[0]
+        lce = entities.LifecycleEnvironment(organization=self.org).create()
+        promote(cvv, lce.id)
+        lce.registry_name_pattern = new_pattern
+        lce = lce.update(['registry_name_pattern'])
+        repos = entities.Repository(organization=self.org).search(
+                query={'environment_id': lce.id})
+
+        expected_pattern = "{}-{}/{}".format(pattern_prefix, self.org.label,
+                                             docker_upstream_name).lower()
+        self.assertEqual(lce.registry_name_pattern, new_pattern)
+        self.assertEqual(repos[0].container_repository_name, expected_pattern)
+
+    @tier2
+    @run_only_on('sat')
+    def test_positive_product_name_change_after_promotion(self):
```

Comment:
not a blocker, just a thought (which might sparkle a fruitful discussion) -
isn't this something that might get parametrized and tried on e.g. all types of repos?

Aren't these tests supposed to go to `test_repository.py` anyway?
I understood `test_docker.py` as a place where the actual integration with docker should be tested.

I can't see how the creating/renaming/updating of the repo challenges the docker integration.
Don't take me wrong, it might as well be me who's confused (and i'm certain that i'm being confused increasingly :))

Reply:
> Aren't these tests supposed to go to `test_repository.py` anyway?

Here's my reasoning for putting them here:
1. Container name pattern is part of larger feature introduced in Sat 6.5. Main idea is that users will be able to issue (part of) Docker API against Sat server. At this point, path under which we store container images will be exposed - and we want to have it configurable.\
I haven't yet get down to writing end to end tests, but I'm pretty sure they fit into this file. I wanted to have everything in one place, so tests for first part of that feature landed here.
2. Similar test cases are defined in this file. See e.g. `test_positive_create_with_name`. You could argue that all of them should be moved, but I'm merely following convention here.
3. This file already has helper function `_create_repository` that is useful for me.

Let me know if that makes sense for you, or you would prefer to move these tests anyway.

As for parametrizing tests - I don't want to turn down discussion, but I think separate issue or our new shiny satqe mailing list is better place to host it. I'm just trying to write my first API tests here :) .

Reply:
i'm actually in favor of keeping the tests organized per feature it's just that AFAIK it's currently not being followed so much elsewhere.
thanks for the reasons you provided.